app/models/review.rb

```diff
@@ -0,0 +1,55 @@
+class Review < ApplicationRecord
+  belongs_to :restaurant
+
+  after_create :add_rating_to_restaurant!
+  after_update :update_rating_to_restaurant!
+  after_destroy :delete_rating_from_restaurant!
+
+  validates :name, :rating, :comment, presence: true
+  validates :name, length: { minimum: 2 }
+  validates :rating, numericality: { greater_than_or_equal_to: WeEat::MIN_REVIEW_RATING,
+                                     less_than_or_equal_to: WeEat::MAX_REVIEW_RATING }
+
+  def add_rating_to_restaurant!
+    review = self
```

Comment:
no need in this assignment, you can later call restaurant without any indication, and it's like calling self.restaurant.